WordPress/gutenberg
# Refactor `useAnchorRef` and related components to work with the new Popover `anchor` prop

## What?


Refactor the return value of the `useAnchorRef` hook, and refactor the components using that hook so that they use the new `Popover`'s `anchor` prop instead of `anchorRef`.

## Why?


See #43691 for more context

## How?


- Change the return value of `useAnchorRef` so that is returns a `VirualElement` instead of a `Range`
- Refactor components using that hook, so that they use the `anchor` prop instead of `anchorRef`

## Testing Instructions





In the editor, test all popovers that are shown as a result of editing inline text (with and without selecting an explicit bit of text), and make sure that:

 - the behavior is the same as on `trunk`
 - there's no warning in the console when any of these popovers are opened

Popovers:

- in a rich text block, select some text and change its formatting (e.g. make it bold)
- in a rich text block, select some text and add an inline link — a popover with an input field to insert a URL should appear next to the selected text
- in a rich text block, select some text and add a "highlight" (via the "⌄" dropdown menu in the toolbar) — a popover with a color picker should appear next to the selected text
- try to edit an image's caption, a popover with a inline toolbar should appear
- in a rich text block, add an inline image, then click on it — a popover allowing you to set the image's width should appear
- in a rich text block, type `@` — a popover allowing you to pick from the list of existing users should appear

**Unit test failures caused by console warnings are expected. The reviews on this PR should focus on the specific refactor to `anchor` prop. This PR will be merged into #43691, so there will be another chance in that PR to give a final review.**
